go.mod

```diff
@@ -19,4 +19,5 @@ require (
 	golang.org/x/crypto v0.0.0-20190820162420-60c769a6c586
 	gopkg.in/yaml.v2 v2.2.8
 	gopkg.in/yaml.v3 v3.0.0-20191010095647-fc94e3f71652
+	k8s.io/apimachinery v0.18.3
```

Comment:
What I have in mind:

```go
type Labels map[string]string

// Matches returns whether all of the given
// key-value pair matches the labels.
// This is to ensure you can "narrow down" by providing more labels
func (l Labels) Matches(list []string) bool {
	matches := true
	for _, s := range list {
		k, v := strings.Split(s)
		l[k]
		matches = matches && l.Match(s)
	}
	return matches
}

func (l Labels) Match(s string) bool {
	k, v := strings.Split(s, "=")
	w, ok := l[k]
	if !ok {
		return false
	}

	//TODO: consider regexp instead
	return w == v
}
```

This obviously is different from `kubectl`, but works very similar to our `-t` (resource filtering) feature.

Above I chose `&&` (`AND`), compared to `||` (`OR`) used by `-t`, so that you can "narrow down" by providing multiple labels. Thoughts?

Reply:
Keep in mind this is just my two cents and I may be totally incorrect on that.

This code just gives me a slightly better feeling, as we can continue to avoid those messy k8s.io dependencies and also keep our existing UX

Reply:
I considered implementing it, but I don't ant the semantics to differ from the kubectl way of using it. And the k8s implementation supports more than just `key=value`. It has things like:

```
environment in (production, qa)
tier notin (frontend, backend)
partition
!partition
```
and it would be nice if those just worked for free if people are used to those.

It isn't including the full client-go packages or anything, it really is a pretty small subset. Probably the largest downside is having to download all of k8s.io/apimachinery to compile, but really I consider that pretty minor.

By using those packages we don't add a new parser that we need to track bugs on and manage divergence from the "official" label selector spec.

Reply:
That's a good point. As there is no binary size penalty here, I am okay with going forward on that package, hoping it doesn't bring us a dependency hell at some point. Fingers crossed :)